app/models/course/assessment/submission.rb

```diff
@@ -10,7 +10,8 @@ class Course::Assessment::Submission < ActiveRecord::Base
 
   workflow do
     state :attempting do
-      event :finalise, transitions_to: :submitted
+      event :finalise, transitions_to: :graded, if: proc { |s| s.automatically_release_grades? }
```

Comment:
Just to follow up, but I will do it in a separate PR: 

@kxmbrian You are right. Currently this implementation works for guided mode, where each answer that is submitted will automatically be auto-graded. 

However, for worksheet mode, answers that are submitted are not automatically auto-graded. So what then happens is that we have a submission that is `graded` with answers that are `submitted` but not `graded`. 

Using an `if` here will not be the best solution. The most probable way is to do some callback, but I'm not sure if I should use the workflow provided one or AR. Will test both then submit a PR then.